File: workshop/c/echo/echo.c
#include <stdio.h>


int main(int argc, char *argv[])
{
    if (argc > 1) {
        int i = 0;
        printf("stdout: ");
        fprintf(stderr, "stderr: ");
        for (i = 1; i < argc; i++) {
            printf("%s ", argv[i]);
            fprintf(stderr, "%s ", argv[i]);
        }
        printf("\n");
        fprintf(stderr, "\n");
    }
    else {
        fprintf(stderr, "You must type a message to echo!\n");
    }

    return 0;
}

File: workshop/python/1-echo/echo.py
import sys


if __name__ == "__main__":
    if len(sys.argv) > 1:
        sys.stdout.write("stdout: ")
        sys.stderr.write("stderr: ")
        for s in sys.argv[1:]:
            sys.stdout.write("%s " % s)
            sys.stderr.write("%s " % s)
        sys.stdout.write("\n")
        sys.stderr.write("\n")
    else:
        sys.stderr.write("You must type a message to echo!\n")
